### DISCUSSION TITLE
DataGrid's SelectedIndex does not update SelectedItem binding.

### DISCUSSION THREAD

**Discussion by Ristellise** [AvaloniaUI/Avalonia#8212] [Q&A]
Hi! As mentioned in the title, After invoking `SelectedIndex` on a DataGrid, it does not trigger a SelectedItem binding.
![image](https://user-images.githubusercontent.com/7894419/170737390-09d0c5b1-21b0-46c0-8226-37d4dcf2cdeb.png)
![image](https://user-images.githubusercontent.com/7894419/170737453-56160778-42aa-4071-8d9c-e8662d720b8f.png)

Sorry, to add on, `EventGridItems` is an `ObservableCollection`
![image](https://user-images.githubusercontent.com/7894419/170739306-b9df4de1-101a-498e-b296-869966cb49a0.png)

**Comment by maxkatz6** [comment]
Hi!
Can you create a repro project so we can test it?
I just tried to make similar code as you have, with my viewmodel, but it works fine:
[AvaloniaTest.zip](https://github.com/AvaloniaUI/Avalonia/files/8809232/AvaloniaTest.zip)

**Reply by Ristellise** [reply]
Hi! I'm not too sure if a repro project is possible but our code is currently on github. <https://github.com/ProjectAmeko/Ameko>

The test repro works fine on my side as well. I have a suspicion it is due to nesting of some sort, but I tried with the `SelectAll()` function and that successfully selected the item as expected.
